File: backend/src/main/java/com/upblit/backend/query/model/Metrics.java
package com.upblit.backend.query.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.time.Instant;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "metrics")
public class Metrics {

    @Id
    private String id;

    @Field("projectId")
    private Long projectId;

    @Field("applicationId")
    private Long applicationId;

    @Field("timestamp")
    private Instant timestamp; // bucket time

    @Field("requestCount")
    private Long requestCount;

    @Field("errorCount")
    private Long errorCount;

    @Field("avgLatency")
    private Double avgLatency;

    @Field("maxLatency")
    private Long maxLatency;

    @Field("minLatency")
    private Long minLatency;
}

File: ingestor/models/Metrics.go
package models

import (
	"time"
)

type Metrics struct {
	ProjectID     int64     `bson:"projectId" json:"projectId"`
	ApplicationID int64     `bson:"applicationId" json:"applicationId"`
	Timestamp     time.Time `bson:"timestamp" json:"timestamp"` // bucket time

	RequestCount int64   `bson:"requestCount" json:"requestCount"`
	ErrorCount   int64   `bson:"errorCount" json:"errorCount"`
	AvgLatency   float64 `bson:"avgLatency" json:"avgLatency"`
	MaxLatency   int64   `bson:"maxLatency" json:"maxLatency"`
	MinLatency   int64   `bson:"minLatency" json:"minLatency"`
}
